### ISSUE TITLE
add internal utility for defining "type" property

### ISSUE THREAD

**Issue by dwhieb** [digitallinguistics/javascript#88]
Add an internal factory method utility for defining the `type` property on a model.

```js
/**
 * A factory function that adds a `type` field to an Object
 * @param  {Object} object The Object to add the `type` field to
 * @param  {String} type   The name of the type to add
 * @return {Object}        Returns the original object
 */
function addType(object, type) {

  Object.defineProperty(object, `type`, {
    configurable: false,
    enumerable:   true,
    get() { return type; },
  });

  return object;

}
```

**Comment by dwhieb** [comment]
This was closed in de93d034bb29f4a49487372b8955c20417faa62e. See `Model.defineTypeProp()`.

**Comment by dwhieb** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [👨🏼‍💻 [javascript] Model.defineTypeProp()](https://trello.com/c/VfkHnfGO/610-%F0%9F%91%A8%F0%9F%8F%BC%F0%9F%92%BB-javascript-modeldefinetypeprop)
